### PR TITLE
[WB-4607] Add wandb artifact cache cleanup command

### PR DESCRIPTION
**Pull request by annirudh**
https://wandb.atlassian.net/browse/WB-4607

Description
-----------

Adds a command to reclaim space from the artifacts cache:

```
wandb artifact cache clean TARGET_SIZE
```

Running it will purge files until we are under `TARGET_SIZE`:

```
$ wandb artifact cache cleanup 10GB
Reclaimed 2GB of space
```

Testing
-------

- Added unit tests
- Tested the command on my local cache directory

### CODE REVIEW

File: wandb/cli/cli.py [hunk 1]
```diff
@@ -1250,11 +1253,58 @@ def human_size(bytes, units=None):
             latest = next(versions)
             print(
                 "{:<15s}{:<15s}{:>15s} {:<20s}".format(
-                    kind.type, latest.updated_at, human_size(latest.size), latest.name
+                    kind.type,
+                    latest.updated_at,
+                    to_human_size(latest.size),
```
Comment (vanpelt) [inline wandb/cli/cli.py:1258]: Not required, but you can use `wandb.util.sizeof_fmt(...)` to do the conversion.  We use it in a bunch of other places.  May be worth througing the reverse logic into wandb.util as well.

**Review by vanpelt — APPROVED** [review]
HOT

File: wandb/cli/cli.py (hunk 1 above)
Comment (annirudh) [reply wandb/cli/cli.py:1258]: Good call. Will do.

File: wandb/sdk/interface/artifacts.py [hunk 2]
```diff
@@ -727,11 +727,36 @@ def get_artifact(self, artifact_id):
     def store_artifact(self, artifact):
         self._artifacts_by_id[artifact.id] = artifact
 
+    def cleanup(self, target_size: int) -> int:
```
Comment (tssweeney) [inline wandb/sdk/interface/artifacts.py:730]: I think this function will leave the directory structure untouched - do we want to potentially cleanup any emptied directories?

**Review by tssweeney — APPROVED** [review]
LGTM - very nice addition. Small optional comment on the cleanup function.

File: wandb/sdk/interface/artifacts.py (hunk 2 above)
Comment (annirudh) [reply wandb/sdk/interface/artifacts.py:730]: I considered it, but figured they don't really take up any space so I didn't really see a pressing need to. No one should be poking around in the cache anyway :)

**Pull request merged**